api/src/main/kotlin/com/google/prefab/api/Android.kt

```diff
@@ -379,30 +380,36 @@ class Android(val abi: Abi, api: Int, val stl: Stl, val ndkMajorVersion: Int) :
         )
     }
 
-    override fun libraryFileFromDirectory(
-        directory: Path,
-        module: Module
-    ): Path = findElfLibrary(directory, module.libraryNameForPlatform(this))
-
     /**
      * The [Android] factory object.
      */
     companion object : PlatformFactoryInterface {
         override val identifier: String = "android"
 
-        override fun fromLibraryDirectory(
-            directory: Path
-        ): PlatformDataInterface {
-            val metadata = Json.decodeFromString<AndroidAbiMetadata>(
-                directory.toFile().resolve("abi.json").readText()
+        override fun prebuiltLibraryFromDirectory(
+            directory: Path, module: Module, loadSchemaVersion: SchemaVersion
+        ): PrebuiltLibrary {
+            val metadata = AndroidAbiMetadata.loadAndMigrate(
+                loadSchemaVersion,
+                directory,
+                module
             )
 
-            return Android(
+            val platformData = Android(
                 Abi.fromString(metadata.abi),
                 metadata.api,
                 Stl.fromString(metadata.stl),
                 metadata.ndk
             )
+            val name = libraryNameFor(module)
+            val ext = if (metadata.isStatic) { "a" } else { "so" }
```

Comment:
(i never liked this syntax in python, but i have to admit i like it even less with mandatory braces [which is in itself a fine idea]. why so much hate for `?:`?)

Reply:
Apparently the braces are optional here. There's a long thread on the Kotlin forums asking why we can't have a ternary operator with the answer being basically "because you don't need it" :) (I agree, it's easier to read for the trivial things than this is)